pkgs/development/python-modules/apache-airflow/default.nix

```diff
@@ -102,6 +122,13 @@ let
     '';
   };
 
+  # Import generated file with metadata for provider dependencies and imports.
+  # Enable additional providers using enabledProviders above.
+  providers = import ./providers.nix;
+  getProviderDeps = provider: map (dep: python.pkgs.${dep}) providers.${provider}.deps;
+  getProviderImports = provider: providers.${provider}.imports;
+  providerDependencies = lib.concatMap getProviderDeps enabledProviders;
+  providerImports = lib.concatMap getProviderImports enabledProviders;
```

Comment:
If we generate the file anyway we can straight away generate a importable file with native nix code.

Reply:
I'm again following the prior art in Home Assistant.  My hope was to minimise the amount of logic generating Nix code from Python, and do as much of it in Nix as possible.